cpp/tests/copying/copy_tests.cu

```diff
@@ -288,4 +288,37 @@ TEST_F(CopyTestUntyped, CopyIfElseTypeMismatch)
 
    EXPECT_THROW(  cudf::experimental::copy_if_else(_lhs, _rhs, _mask),
                   cudf::logic_error);
-}
\ No newline at end of file
+}
+
+struct StringsCopyIfElseTest : public cudf::test::BaseFixture {};
+
+struct filter_test_fn
+{
+    __host__ __device__ bool operator()(cudf::size_type idx) const
+    {
+        return static_cast<bool>(idx % 2);
+    }
+};
+
+TEST_F(StringsCopyIfElseTest, CopyIfElse)
+{
+    std::vector<const char*> h_strings1{ "eee", "bb", "", "aa", "bbb", "ééé" };
+    cudf::test::strings_column_wrapper strings1( h_strings1.begin(), h_strings1.end() );
+    auto lhs = cudf::strings_column_view(strings1);
+    std::vector<const char*> h_strings2{ "zz",  "", "yyy", "w", "ééé", "ooo" };
+    cudf::test::strings_column_wrapper strings2( h_strings2.begin(), h_strings2.end() );
+    auto rhs = cudf::strings_column_view(strings2);
+
+    auto results = cudf::strings::detail::copy_if_else(lhs,rhs,filter_test_fn{});
```

Comment:
Lambda is cleaner

```suggestion
    auto filter = [] __device__ (auto idx) { return idx % 2; };
    auto results = cudf::strings::detail::copy_if_else(lhs,rhs,filter);
```

Reply:
I tried that before. It will not compile. Something weird in the gtest environment:
```
error: The enclosing parent function ("TestBody") for an extended __device__ lambda
cannot have private or protected access within its class
```

Reply:
Ah yes, device lambda limit because gtests body is defined as a private function. No worries.